articles/views.py

```diff
@@ -1,16 +1,75 @@
-from django.shortcuts import get_object_or_404
+from django.contrib.auth.mixins import LoginRequiredMixin
 from django.urls import reverse_lazy
-from django.shortcuts import render
+from django.views.generic import DeleteView, DetailView, TemplateView
 from django.http import HttpResponseRedirect
-from django.contrib.auth.decorators import login_required
-from django.contrib.auth.mixins import LoginRequiredMixin
 from django.views.generic.edit import CreateView, UpdateView
-from .models import Article
-from .forms import ArticleForm
 
 
-def view(req, slug):
-    return render(req, "articles/detail.html")
+from .models import Article, Comment
+from .forms import ArticleForm, CommentForm
+from .mixins import UserIsAuthorMixin
+
+
+class ViewArticle(DetailView):
+    model = Article
+    template_name = "articles/detail.html"
+    context_object_name = "article"
+
+
+class AddComment(LoginRequiredMixin, CreateView):
+    model = Comment
+    template_name = "articles/_add_comment.html"
+    form_class = CommentForm
+
+    def get_login_url(self):
+        return reverse_lazy("login_to_add_comment", kwargs={"slug": self.kwargs["slug"]})
+
+    def get_success_url(self):
+        return reverse_lazy("article_view", kwargs={"slug": self.kwargs["slug"]})
```

Comment:
Instead of redirecting, can we also return a `prepend` stream response? That way we don't have to do a page refresh and the scroll position isn't lost.

Reply:
Yea, I was just looking at that too. I was wondering how much we want to do before we have a package to add and extract things with.

Reply:
Our main purpose here is to find patterns in real apps to abstract with a package. We'd want to make all features in this app as idiomatic as possible so we can do that abstraction later once we see what the patterns are. I think this feature in particular is one that shows up a lot and is a great example to explore the ideal implementation with.

Reply:
Good point, it's certainty a scenario that appears regularly.